src/test/java/org/broadinstitute/dsde/consent/ontology/datause/builder/ConsentRestrictionBuilderTest.java

```diff
@@ -11,73 +12,155 @@
 public class ConsentRestrictionBuilderTest {
 
     private UseRestrictionBuilder restrictionBuilder = new ConsentRestrictionBuilder();
+    private UseRestriction everything = new Everything();
 
     @Test
     public void testGeneralUse() {
         DataUse dataUse = new DataUse();
         dataUse.setGeneralUse(true);
         UseRestriction restriction = restrictionBuilder.buildUseRestriction(dataUse);
-        Assert.assertTrue(restriction.equals(new Everything()));
+        Assert.assertEquals(restriction, everything);
+    }
+
+    /**
+     * All of the general use mixed cases are to ensure that GRU is ignored when
+     * sub-conditions exist that would render it as a non-GRU use restriction
+     */
+    @Test
+    public void testGeneralUseMixedCase1() {
+        DataUse dataUse = new DataUseBuilder().
+                setGeneralUse(true).
+                setDiseaseRestrictions(Collections.singletonList(ConsentUseCases.CANCER)).
+                build();
+        UseRestriction restriction = restrictionBuilder.buildUseRestriction(dataUse);
+        Assert.assertNotEquals(restriction, everything);
```

Comment:
I'm bothered by `assertNotEquals`. I've seen tests succeed with obviously wrong behavior because they used `assertNotEquals`. Can you instead check for what the result should be instead of what it shouldn't be?

Reply:
We have lots of tests that already test for what a restriction should be. This test is specifically for what it should not be.

Reply:
If there are tests for what a result should be then a separate test for what it should not be is redundant. Since these were the added tests (the rest of the changes in this file look like refactoring), I figured this was important coverage. Shouldn't there be other tests that have to change their expectations if the behavior is changing?

I'll also point out that I'm a bit out of my domain here so the coverage may be there but is not obvious to me.

Reply:
The original code made the assumption that if a `generalUse` flag was present, then the restriction would *always* be of that type. The requirements have changed such that we need to ignore that flag when other sub conditions exist. 

The existing tests all worked for the expected cases, and we never needed to check for this new case. They all still work even with the new logic flow, which is the most critical point.

These new tests are only for the singular purpose of the new case. They can certainly be considered redundant, but I'd rather have them and make sure the new assumptions still hold true rather than not having them at all.

Reply:
One more point to help clarify the point of our current tests. We have a extensive set of use cases that require specific outcomes for various combinations of choices. Those are each carefully constructed and tested for such that we get the right results when trying to match a research purpose and a consent. The test cases all map to something very specific. 

For the new test cases, it really doesn't matter what the end result of the restriction ends up being - we have no requirements for what a `generalUse` + `commercialUse` restriction needs to looks like. The only requirement we have is that it not be of `{"type":"everything"}`

Reply:
@breilly2 - I've thought about this some more and I think I can address your questions. I will take a stab at something more specific for these tests.